src/libraries/System.Text.Json/tests/Serialization/ReferenceHandlingTests.Deserialize.cs

```diff
@@ -1258,8 +1262,8 @@ public static void PreservedArrayWithoutValues()
             }";
             JsonException ex = Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<List<int>>(json, s_deserializerOptionsPreserve));
 
-            // Not sure if is ok for Path to have this value.
-            Assert.Equal("$.$id", ex.Path);
+            Assert.Equal("$", ex.Path);
```

Comment:
So when `$values` is missing we point back to the root?

Reply:
We don't want to include a previously good property name in the JsonPath, at least as far as I'm aware. So we don't show a property name in those cases. E.g. `"{"MyProp":1, blah}"` would fail with JsonPath=`$` not JsonPath=`$.MyProp`